URBNSwiftAlert/Classes/AlertStyler.swift

```diff
@@ -0,0 +1,448 @@
+         */
+        public var titleColor = UIColor.black
+        
+        /**
+         * Background color of the buttons for active alerts
+         */
+        public var backgroundColor = UIColor.gray
+        
+        /**
+         * Background color of a highlighted button for an active alert
+         */
+        public var highlightBackgroundColor = UIColor.darkGray
+        
+        /**
+         * Background color of a selected button for an active alert
+         */
+        public var selectedBackgroundColor = UIColor.lightGray
+        
+        /**
+         * Button title color for a selected state
+         */
+        public var selectedTitleColor = UIColor.black
+        
+        /**
+         * Button container inset constraints
+         */
+        public var containerInsetConstraints = InsetConstraints(insets: UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10), priority: UILayoutPriorityDefaultHigh)
+        
+        public var spacing: CGFloat = 10.0 {
+            didSet {
+                assert(spacing > 0, "Button spacing must be greater than 0")
+            }
+        }
+        
+        /**
+         * Corner radius of the alert's buttons
+         */
+        public var cornerRadius: CGFloat = 0.0
+        
+        /**
+         * Font of the button's titles
+         */
+        public var font: UIFont? = UIFont.systemFont(ofSize: 12.0)
+        
+        /**
+         * Button title color on highlight
+         */
+        public var highlightTitleColor = UIColor.black
+        
+        /**
+         * UIEdgeInsets used at the external margins for the buttons of the alert's buttons
+         */
+        public var insetConstraints = InsetConstraints(insets: UIEdgeInsets.zero, priority: UILayoutPriorityDefaultHigh)
+        
+        /**
+         * Width of the alert's button's border layer
+         */
+        public var borderWidth: CGFloat = 0.0
+        
+        /**
+         * UIColor of the alert's button's border
+         */
+        public var borderColor = UIColor.clear
+        
+        /**
+         * Opacity of the alert's button's shadows
+         */
+        public var shadowOpacity: Float = 0.0
+        
+        /**
+         * Radius of the alert's button's shadows
+         */
+        public var shadowRadius: CGFloat = 0.0
+        
+        /**
+         * Color of the alert's button's shadows
+         */
+        public var shadowColor = UIColor.clear
+        
+        /**
+         * Offset of the alert's button's shadows
+         */
+        public var shadowOffset = CGSize.zero
+        
+        /**
+         * Insets of the buttons contents
+         */
+        public var contentInsets = UIEdgeInsets.zero
+        
+        /**
+         *  Layout axis for buttons (for 3+ always vertical always used)
+         */
+        public var layoutAxis = UILayoutConstraintAxis.horizontal
+    }
+    
+    public struct CancelButton {
```

Comment:
These buttons seem like they should share interfaces with the `Button` type.

Reply:
im not happy with the button architecture.  conventions such as a cancel button or destructive button seem unnecessary as users just define their own custom ones anyway. 

 - some options ive considered are -
1) completely remove the different types of buttons (cancel, destructive, disabled) and just have 1 generic button type
2) fold the types (cancel/destructive/disable) under the button struct as enums

Reply:
I think either one of those options would be preferred. I was thinking (1) myself, but (2) makes sense as it’s fairly lightweight.